test/transactions/deleverage.ts

```diff
@@ -0,0 +1,73 @@
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import * as api from 'test/fixtures/api';
+import { claimToken, getChainId, polygonTokens, snapshotAndRevertOnce } from '@protocolink/test-helpers';
+import * as common from '@protocolink/common';
+import { expect } from 'chai';
+import hre from 'hardhat';
+import * as logics from '@protocolink/logics';
+import * as utils from 'test/utils';
+
+describe('Transaction: Deleverage', function () {
+  let chainId: number;
+  let user: SignerWithAddress;
+
+  before(async function () {
+    chainId = await getChainId();
+    [, user] = await hre.ethers.getSigners();
+    await claimToken(chainId, user.address, polygonTokens.WETH, '5');
+  });
+
+  snapshotAndRevertOnce();
+
+  it('user deleverage his USDC market position with WETH', async function () {
+    const marketId = logics.compoundv3.MarketId.USDC;
+
+    // 1. user has supplied 5 WETH
+    const supplyAmount = new common.TokenAmount(polygonTokens.WETH, '5');
+    await utils.supply(chainId, user, marketId, supplyAmount);
+
+    // 2. user has borrowed 2000 USDC
+    const borrowAmount = new common.TokenAmount(polygonTokens.USDC, '2000');
+    await utils.borrow(chainId, user, marketId, borrowAmount);
+
+    // 3. user obtains a quotation for deleveraging the total debt through the deleverage API
+    const slippage = 100;
+    const deleverageCollateralToken = polygonTokens.WETH;
+    const deleverageAmount = new common.TokenAmount(borrowAmount.token).setWei(
+      common.calcSlippage(borrowAmount.amountWei, -slippage)
+    );
+    const quotation = await api.quote(chainId, marketId, 'deleverage', {
+      account: user.address,
+      token: deleverageCollateralToken,
+      amount: deleverageAmount.amount,
```

Comment:
Do we need to add a test to verify the scenario: "The deleverage amount is greater than the current debt position"?

Reply:
I think the case is tested in the unit test. I can still add it if you think it's necessary.

Reply:
hmm, I'm not sure I found one in our unit test since we cannot preset a debt position. The scenario would be having USDC debt position (say 2000) and deleverage value (say 4000 USDC) but only repaying 2000 USDC in our logic.

Reply:
Ah got it. I misunderstood what you mentioned. The current test case is indeed deleveraging an amount that is greater than the debt position. What we didn't test here is deleveraging an amount that is exactly the same as the debt position (which is not possible since that the debt keeps accumulating the interest) and an amount that is less than the debt position. The latter one is not doable because currently we are not able to make the api being updated after building the position, which makes the logic become repaying all instead of partial repay.